query/src/interpreters/interpreter_insert.rs

```diff
@@ -160,15 +161,24 @@ impl InsertInterpreter {
         executor.execute()?;
         drop(executor);
 
+        let overwrite = self.plan.overwrite;
+        let catalog_name = self.plan.catalog_name.clone();
+        let context = self.ctx.clone();
         let append_entries = self.ctx.consume_precommit_blocks();
-        table
-            .commit_insertion(
-                self.ctx.clone(),
-                &self.plan.catalog_name,
-                append_entries,
-                self.plan.overwrite,
-            )
-            .await?;
+        let handler = self.ctx.get_storage_runtime().spawn(async move {
```

Comment:
We should add some comments to tell why we need a `spawn` here :D

Reply:
Sorry, I have added and committed comment, but I forgot to push : (

Reply:
We can do it in another PR :)